test/cli/pin.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-env mocha */
+'use strict'
+
+const expect = require('chai').expect
+const series = require('async/series')
+const runOnAndOff = require('../utils/on-and-off')
+const _ = require('lodash')
+
+// use a tree of ipfs objects for recursive tests:
+//  root
+//   |`leaf
+//    `branch
+//      `subLeaf
+
+const filenames = [ 'root', 'leaf', 'branch', 'subLeaf' ]
+
+const keys = {
+  root: 'QmWQwS2Xh1SFGMPzUVYQ52b7RC7fTfiaPHm3ZyTRZuHmer',
+  leaf: 'QmaZoTQ6wFe7EtvaePBUeXavfeRqCAq3RUMomFxBpZLrLA',
+  branch: 'QmNxjjP7dtx6pzxWGBRCrgmjX3JqKL7uF2Kjx7ExiZDbSB',
+  subLeaf: 'QmUzzznkyQL7FjjBztG3D1tTjBuxeArLceDZnuSowUggXL'
+}
+
+describe('pin', () => runOnAndOff((thing) => {
```

Comment:
If you want to test only "offline" do `runOnAndOff.off`

If you want to run just these tests, do: `describe.only`